index.js

```diff
@@ -235,6 +241,10 @@ function addPreflightHeaders (req, reply, corsOptions) {
   if (corsOptions.maxAge !== null) {
     reply.header('Access-Control-Max-Age', String(corsOptions.maxAge))
   }
+
+  if (corsOptions.cacheControl && (typeof corsOptions.cacheControl === 'string')) {
```

Comment:
@jsumners 

do we still need a type check?

```suggestion
  if (corsOptions.cacheControl) {
```

Reply:
Yeah, the normalization has cleared this behaviour

cc @brettwillis

Reply:
@Eomm are you agreeing to apply the suggested change? Initially we had `cacheControl` only applied if the value was `string` or `number`. If we remove the type check then we could end up applying any non-`number` value. 

We should either (1) keep the check, (2) coerce the "any truthy value" to string e.g. `String(cacheControl)` and apply it, or (3) patch normalisation to also set `cacheControl` to `null` if it is not a `string` (and not a `number`).

Reply:
This is-string check needs to happen in order to not add an invalid header value (still possible, but the module has done as much as it really can). Test should be added to show this if they do not already exist.

Reply:
> Test should be added to show this if they do not already exist.

This, or just the `normalizeCorsOptions` do the full normalization :D

This check protects us if the user provides a function or and object.
Right now - it is just ignored. Let's add a test and it is fine too

Reply:
I discovered that this `Number.isInteger(Number(corsOptions.cacheControl)) ` behaviour means that a `boolean` values are coerced to `max-age=1` and `max-age=0` respectively, which is probably not desired.

Should we add a check to exclude `boolean`s e.g. `Number.isInteger(Number(corsOptions.cacheControl)) && (typeof corsOptions.cacheControl !== 'boolean')`?

Or only consider values that are strictly a `number` (don't coerce to number) e.g. `Number.isInteger(corsOptions.cacheControl)`?

Reply:
Either accept a string or an integer. Do not accept anything else.

Reply:
> Either accept a string or an integer. Do not accept anything else.

Ok, done.

Also I moved the is-string check to the normalisation function, so there is one less conditional check being run per-request.